unconvert.go

```diff
@@ -386,35 +371,32 @@ func (v *visitor) unconvert(call *ast.CallExpr) {
 	if len(call.Args) != 1 || call.Ellipsis != token.NoPos {
 		return
 	}
-	ft, ok := v.pkg.Types[call.Fun]
-	if !ok {
+	ft := v.pkg.TypesInfo.TypeOf(call.Fun)
+	if ft == nil {
 		fmt.Println("Missing type for function")
 		return
 	}
-	if !ft.IsType() {
-		// Function call; not a conversion.
```

Comment:
Why did you get rid of this?

Reply:
To elaborate, I think this is necessary to prevent unconvert from mishandling edge cases like:

    type T func(T)
    var f T
    _ = f(f)

Without checking that `call.Fun` is a type expression, I expect unconvert would rewrite the last line into just `_ = f`.

Reply:
Good question. I honestly cannot exactly remember at this point anymore. But what I can gather from context is that, `TypesInfo.TypeOf` should always return a type. But I'll verify tomorrow.

Reply:
TypeOf will always return a type, yes. But this check was distinguishing whether an expression was a value or a type. This is important for distinguishing `functionName(x)` (function call) from `typeName(x)` (conversion).

Reply:
Fixed